src/views/Entities/Zones.vue

```diff
@@ -282,214 +299,236 @@ export default {
+          });
+      };
 
-      getZoneStatus()
+      getZoneStatus();
     },
-    getEntity (entity) {
-      this.entityIsLoading = true
-      this.entityIsEmpty = false
+    getEntity(entity) {
+      this.entityIsLoading = true;
+      this.entityIsEmpty = false;
 
-      const selected = ['type', 'name', 'mesh']
+      const selected = ['type', 'name', 'mesh'];
 
       const promise = new Promise((resolve, reject) => {
         if (entity && entity !== null) {
-          resolve(entity)
+          resolve(entity);
         } else {
-          const error = new Error('Entity is either undefined or not present.')
+          const error = new Error('Entity is either undefined or not present.');
 
-          reject(error)
+          reject(error);
         }
-      })
+      });
 
       if (entity && entity !== null) {
         promise
-          .then(response => {
+          .then((response) => {
             if (response) {
               // get the Zone details from the Zone Insights endpoint
-              this.$api.getZoneOverview(response.name)
-                .then((response) => {
-                  this.tabGroupTitle = `Zone: ${response.name}`
-                  this.entityOverviewTitle = `Zone Overview for ${response.name}`
-
-                  // remove `zoneInsight` from the response since we already
-                  // use it in the Zone Insights tab
-                  const cleanRes = () => {
-                    const src = Object.assign({}, response)
-
-                    delete src.zoneInsight
-
-                    return src
-                  }
-
-                  this.entity = getSome(response, selected)
-                  this.rawEntity = stripTimes(response)
-                  this.yamlEntity = cleanRes()
-                })
+              this.$api.getZoneOverview(response.name).then((response) => {
+                this.tabGroupTitle = `Zone: ${response.name}`;
+                this.entityOverviewTitle = `Zone Overview for ${response.name}`;
+
+                let remoteCpVersion = '-';
+                /**
+                 * Iterate through the subscriptions
+                 */
+                if (
+                  response.zoneInsight.subscriptions &&
+                  response.zoneInsight.subscriptions.length
+                ) {
+                  response.zoneInsight.subscriptions.forEach((item) => {
+                    if (item.version && item.version.kumaCp) {
+                      remoteCpVersion = item.version.kumaCp.version;
+                    }
+                  });
+                }
+
+                // remove `zoneInsight` from the response since we already
+                // use it in the Zone Insights tab
+                const cleanRes = () => {
+                  const src = Object.assign({}, response);
+
+                  delete src.zoneInsight;
+
+                  return src;
+                };
+                if (response) {
```

Comment:
This check id unnecessary as it's already checked in the line 464, and also if it would be not true, it would crash in lines 475-476